### DISCUSSION TITLE
Add option to view how much both the app's size + it's data takes up in the storage when viewing the app's info

### DISCUSSION THREAD

**Discussion by AkiraKusanagi** [EmiyaSyahriel/CrossLauncher#54] [Ideas]
When viewing the info of an app, it would seem a bit better to view how much an app's size plus how much it's data takes up in the storage because there are some apps where installing them will take up very little space, but then need to install like 3gb+ of data.

**Comment by AkiraKusanagi** [comment]
For example, Ace Attorney Investigations: Miles Edgeworth (Or Gyakuten Kenji), the app takes up 27.18mb, but it has data (Which needs to be installed) that is about 226mb. If you view the info of the app, it would just show the app's size. It would be better to see how much the ENTIRE app (Size + Data), takes up. (Somewhere around 253mb.)

**Reply by EmiyaSyahriel** [reply]
It's impossible for CrossLauncher, the permission to look around other app's data (including measuring it's total size), AFAIK is only for System apps, other apps are not normally permitted to do it. The best I can get is total app's package size (including splitted packages) since it's public for other apps, while the app's data is private to the app itself (even if it is in shared storage) and some other apps like File Managers.

There might be some apps that is able to do the total data size measuring, but I doubt they are stable enough in recent Android version because they usually uses hacks or because of looser permission of older Android version, or they will just ask for permission everytime the app changes selection (which is apparently the intended way by the Android security designer).

But, if you found an open-source Android app that is capable to measure another app's total size without asking permission everytime or without needing to become System apps and is still stable in newer Android version, you can put the link here, I might be able to copy their implementation here.

**Reply by EmiyaSyahriel** [reply]
Never mind, found the API : [Usage Access Permission](https://stackoverflow.com/questions/38686632/how-to-get-usage-access-permission-programmatically) & [Manage All Files Permission](https://developer.android.com/training/data-storage/manage-all-files)
